src/pm/hydra2/mpiexec/mpiexec.c

```diff
@@ -835,14 +847,23 @@ int main(int argc, char **argv)
     args[i++] = HYD_str_from_int(mpiexec_params.usize);
     args[i++] = NULL;
 
+    int time_left = HYD_get_time_left(&mpiexec_params.timeout);
+
     status =
         HYD_bstrap_setup(mpiexec_params.base_path, mpiexec_params.launcher,
                          mpiexec_params.launcher_exec, pg->node_count, pg->node_list, -1,
                          mpiexec_params.port_range, args, 0, &pg->num_downstream,
                          &pg->downstream.fd_stdin, &pg->downstream.fd_stdout_hash,
                          &pg->downstream.fd_stderr_hash, &pg->downstream.fd_control_hash,
                          &pg->downstream.proxy_id, &pg->downstream.pid, mpiexec_params.debug,
-                         mpiexec_params.tree_width);
+                         mpiexec_params.tree_width, time_left);
+
+    if (status == HYD_ERR_TIMED_OUT) {
```

Comment:
This block reads odd to me. We check for timeout, and then call `HYD_check_timed_out` presumably to print the timeout message?

Reply:
Fixed